app/utility/base_planning_svc.py

```diff
@@ -140,3 +142,21 @@ def _is_fact_bound(fact):
             if not requirement.enforce(combo[0], uf, operation.all_facts()):
                 return False
         return True
+
+    async def _exclude_existing(self, combo, operation):
+        all_hostnames = [agent.paw.split('$')[0].lower() for agent in await self._active_agents(operation)]
```

Comment:
Henry is putting in a change that will make this easier and cleaner. See his or out to add host and username as agent attributes

Reply:
We've talked about it. If his code gets in before this, I'll update it accordingly.